{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d5ac18fe",
   "metadata": {},
   "source": [
    "# M01(VL01) Daiyun Dong"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a32b482",
   "metadata": {},
   "source": [
    "### Feature\n",
    "#### 7PFL\n",
    "```\n",
    "loop_\n",
    "_database_2.database_id \n",
    "_database_2.database_code \n",
    "_database_2.pdbx_database_accession \n",
    "_database_2.pdbx_DOI \n",
    "PDB   7PFL         pdb_00007pfl 10.2210/pdb7pfl/pdb \n",
    "WWPDB D_1292117595 ?            ?   \n",
    "```\n",
    "> This section appears at the beginning of the document and serves to provide an overview of the protein information. It indicates that the protein is 7PFL, with the accession number pdb_00007pfl and DOI 10.2210/pdb7pfl/pdb. Additionally, its wwPDB identifier is D_1292117595.\n",
    "\n",
    "```\n",
    "_symmetry.entry_id                         7PFL \n",
    "_symmetry.cell_setting                     ? \n",
    "_symmetry.Int_Tables_number                5 \n",
    "_symmetry.space_group_name_Hall            ? \n",
    "_symmetry.space_group_name_H-M             'C 1 2 1' \n",
    "_symmetry.pdbx_full_space_group_name_H-M   ? \n",
    "```\n",
    "> This section indicates that the 7PFL protein crystal belongs to space group number 5 and the 'C121' space group.\n",
    "\n",
    "```\n",
    "loop_\n",
    "_entity.id \n",
    "_entity.type \n",
    "_entity.src_method \n",
    "_entity.pdbx_description \n",
    "_entity.formula_weight \n",
    "_entity.pdbx_number_of_molecules \n",
    "_entity.pdbx_ec \n",
    "_entity.pdbx_mutation \n",
    "_entity.pdbx_fragment \n",
    "_entity.details \n",
    "1 polymer     man 'Replicase polyprotein 1ab' 33825.547 1   ? ? ? ? \n",
    "2 non-polymer syn 'SODIUM ION'                22.990    1   ? ? ? ? \n",
    "3 water       nat water      \n",
    "```\n",
    "> This section indicates that entity 1 is an artificial polymer (identified from 'src_method'), with a molecular weight (identified from 'formula_weight') of 33,825.547 Daltons. Entity 2 is a synthetic sodium ion, with a molecular weight of 22.990 Daltons. Entity 3 is a water molecule.\n",
    "\n",
    "```\n",
    "_exptl.method                     'X-RAY DIFFRACTION' \n",
    "_exptl_crystal.density_Matthews            2.01 \n",
    "_exptl_crystal.density_percent_sol         38.68 \n",
    "_exptl_crystal_grow.method          'VAPOR DIFFUSION, SITTING DROP' \n",
    "_exptl_crystal_grow.temp            293 \n",
    "```\n",
    "> Organize useful data related to the experiment. 'method' indicates that the test method was X-ray diffraction. 'density_Matthews' indicates that the protein density within the crystals was 2.01 Å³/Da. 'density_percent_sol' indicates that the solvent percentage is 38.68%. 'grow.method' indicates that the protein crystals were grown by vapor diffusion and static droplet method. 'grow.temp ' indicates that the protein crystal growth temperature is 293K.\n",
    "\n",
    "```\n",
    "_diffrn.ambient_temp                     100 \n",
    "_diffrn_detector.detector                     PIXEL \n",
    "_diffrn_detector.type                         'DECTRIS EIGER X 16M' \n",
    "_diffrn_detector.pdbx_collection_date         2020-06-27  \n",
    "_diffrn_radiation.pdbx_monochromatic_or_laue_m_l   M \n",
    "_diffrn_radiation.pdbx_diffrn_protocol             'SINGLE WAVELENGTH' \n",
    "_diffrn_radiation.pdbx_scattering_type             x-ray \n",
    "_diffrn_radiation_wavelength.wavelength   0.9763 \n",
    "_diffrn_radiation_wavelength.wt           1.0 \n",
    "_diffrn_source.source                      SYNCHROTRON \n",
    "_diffrn_source.type                        'MAX IV BEAMLINE BioMAX' \n",
    "_diffrn_source.pdbx_wavelength_list        0.9763 \n",
    "_diffrn_source.pdbx_synchrotron_beamline   BioMAX \n",
    "_diffrn_source.pdbx_synchrotron_site       'MAX IV' \n",
    "```\n",
    "> Organize useful data related to the experiment.'ambient_temp' indicates that the X-ray diffraction experiment was conducted, and temperature is 100K. 'detector' indicates that the detector was a PIXEL, and 'pdbx_collection_date' indicates a date of 06/27/2020. 'monochromatic_or_laue_m_l' indicates that monochromatic X-rays were used at a wavelength of 0.9763 Å, and the radiation source was synchrotron.\n",
    "\n",
    "```\n",
    "_reflns.d_resolution_high                              1.800 \n",
    "_reflns.d_resolution_low                               34.020 \n",
    "_reflns.number_obs                                     24173 \n",
    "_reflns.percent_possible_obs                           97.300 \n",
    "_reflns.pdbx_redundancy                                3.500 \n",
    "_reflns.pdbx_Rmerge_I_obs                              0.097 \n",
    "_reflns.pdbx_netI_over_sigmaI                          7.200 \n",
    "_reflns.pdbx_scaling_rejects                           139 \n",
    "_reflns.pdbx_Rrim_I_all                                0.115 \n",
    "_reflns.pdbx_Rpim_I_all                                0.062  \n",
    "_reflns.pdbx_number_measured_all                       84326 \n",
    "_reflns.pdbx_CC_half                                   0.990  \n",
    "```\n",
    "> Organize useful data related to the experiment. The section indicates a high resolution of 1.800 Å, a low resolution of 34.020 Å, a diffraction point number of 24,173, a percentage of possible observed reflections of 97.300%, a redundancy of 3.500, an Rmerge value of 0.097, and a ratio of the intensity of the reflections to their standard deviation of 7.200. The number of scaling rejections is 139, the Rrim value is 0.115, the Rpim value is 0.062, the total number of reflections measured is 84326, and the semi-correlation coefficient is 0.990.The above values are used as a measure of the data quality of the experiment and are used to assess the reliability of this data.\n",
    "\n",
    "#### 5ZMD\n",
    "```\n",
    "loop_\n",
    "_database_2.database_id \n",
    "_database_2.database_code \n",
    "_database_2.pdbx_database_accession \n",
    "_database_2.pdbx_DOI \n",
    "PDB   5ZMD         pdb_00005zmd 10.2210/pdb5zmd/pdb \n",
    "WWPDB D_1300007324 ?            ? \n",
    "```\n",
    "> This section introduce the basic information about this protein structure. The protein id is 5ZMD, the accession is pdb_00005zmd and doi is 10.2210/pdb5zmd/pdb.\n",
    "\n",
    "```\n",
    "_symmetry.entry_id                         5ZMD \n",
    "_symmetry.cell_setting                     ? \n",
    "_symmetry.Int_Tables_number                24 \n",
    "_symmetry.space_group_name_Hall            ? \n",
    "_symmetry.space_group_name_H-M             'I 21 21 21' \n",
    "_symmetry.pdbx_full_space_group_name_H-M   ? \n",
    "```\n",
    "> This section indicates the information about the protein crystal. It belongs to space group number 24 and the 'I 21 21 21' space group.\n",
    "\n",
    "```\n",
    "loop_\n",
    "_entity.id \n",
    "_entity.type \n",
    "_entity.src_method \n",
    "_entity.pdbx_description \n",
    "_entity.formula_weight \n",
    "_entity.pdbx_number_of_molecules \n",
    "_entity.pdbx_ec \n",
    "_entity.pdbx_mutation \n",
    "_entity.pdbx_fragment \n",
    "_entity.details \n",
    "1 polymer     man 'Alpha-ketoglutarate-dependent dioxygenase FTO'  53312.223 4  1.14.11.- 'Q86K, Q306K' ? ? \n",
    "2 polymer     syn \n",
    ";DNA (5'-D(P*TP*CP*TP*(6MA)P*TP*AP*TP*CP*G)-3')\n",
    ";\n",
    "2719.823  4  ?         ?             ? ? \n",
    "3 non-polymer syn 'MANGANESE (II) ION'                             54.938    4  ?         ?             ? ? \n",
    "4 non-polymer syn N-OXALYLGLYCINE                                  147.086   4  ?         ?             ? ? \n",
    "5 water       nat water                                            18.015    18 ?         ?             ? ? \n",
    "```\n",
    "> This section indicates the entities in this record. The entity 1 is the protein with a MW of 53312.223 Daltons. The entity 2 is the DNA with a MW of 2719.823 Daltons. The entity 3 is the manganese ion with a MW of 54.938 Daltons. The entity 4 is the N-oxalylglycine with a MW of 147.086 Daltons. The entity 5 is the water with a MW of 18.015 Daltons.\n",
    "\n",
    "```\n",
    "_entity_src_gen.entity_id                          1 \n",
    "_entity_src_gen.pdbx_src_id                        1 \n",
    "_entity_src_gen.pdbx_alt_source_flag               sample \n",
    "_entity_src_gen.pdbx_seq_type                      'Biological sequence' \n",
    "_entity_src_gen.pdbx_beg_seq_num                   1 \n",
    "_entity_src_gen.pdbx_end_seq_num                   463 \n",
    "_entity_src_gen.gene_src_common_name               Human \n",
    "_entity_src_gen.pdbx_gene_src_gene                 'FTO, KIAA1752' \n",
    "_entity_src_gen.pdbx_gene_src_scientific_name      'Homo sapiens' \n",
    "_entity_src_gen.pdbx_gene_src_ncbi_taxonomy_id     9606 \n",
    "_entity_src_gen.pdbx_host_org_scientific_name      'Escherichia coli' \n",
    "_entity_src_gen.pdbx_host_org_ncbi_taxonomy_id     562 \n",
    "```\n",
    "> This section describes where the protein(entity 1) comes from. The gene name of entity 1 is FTO(KIAA1752) and is from human. The host of this protein is E.coli.\n",
    "\n",
    "```\n",
    "_pdbx_entity_src_syn.entity_id              2 \n",
    "_pdbx_entity_src_syn.pdbx_src_id            1 \n",
    "_pdbx_entity_src_syn.pdbx_alt_source_flag   sample \n",
    "_pdbx_entity_src_syn.pdbx_beg_seq_num       1 \n",
    "_pdbx_entity_src_syn.pdbx_end_seq_num       9 \n",
    "_pdbx_entity_src_syn.organism_scientific    'DNA launch vector pDE-GFP2' \n",
    "_pdbx_entity_src_syn.organism_common_name   ? \n",
    "_pdbx_entity_src_syn.ncbi_taxonomy_id       415098 \n",
    "_pdbx_entity_src_syn.details                ? \n",
    "```\n",
    "> This section describes where the DNA(entity 2) comes from. The organism is the DNA launch vector pDE-GFP2.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6a76b5d3",
   "metadata": {},
   "source": [
    "### Atomic-level Data\n",
    "#### 7PFL\n",
    "\n",
    "```\n",
    "loop_\n",
    "_atom_site.group_PDB \n",
    "_atom_site.id \n",
    "_atom_site.type_symbol \n",
    "_atom_site.label_atom_id \n",
    "_atom_site.label_alt_id \n",
    "_atom_site.label_comp_id \n",
    "_atom_site.label_asym_id \n",
    "_atom_site.label_entity_id \n",
    "_atom_site.label_seq_id \n",
    "_atom_site.pdbx_PDB_ins_code \n",
    "_atom_site.Cartn_x \n",
    "_atom_site.Cartn_y \n",
    "_atom_site.Cartn_z \n",
    "_atom_site.occupancy \n",
    "_atom_site.B_iso_or_equiv \n",
    "_atom_site.pdbx_formal_charge \n",
    "_atom_site.auth_seq_id \n",
    "_atom_site.auth_comp_id \n",
    "_atom_site.auth_asym_id \n",
    "_atom_site.auth_atom_id \n",
    "_atom_site.pdbx_PDB_model_num \n",
    "ATOM   1    N  N   . SER A 1 1   ? -3.231 -7.482  -17.009 1.00 40.61  ? 1   SER A N   1 \n",
    "ATOM   2    C  CA  . SER A 1 1   ? -2.514 -6.334  -16.473 1.00 32.76  ? 1   SER A CA  1 \n",
    "ATOM   3    C  C   . SER A 1 1   ? -2.760 -6.189  -14.976 1.00 38.09  ? 1   SER A C   1 \n",
    "ATOM   4    O  O   . SER A 1 1   ? -3.347 -7.070  -14.346 1.00 37.15  ? 1   SER A O   1 \n",
    "```\n",
    "\n",
    "> Each row in the penultimate 4 rows corresponds to an atom. The top 21 rows explain the meaning of each of the following columns. As an example, in the penultimate four rows, the first column, ATOM, indicates that the atom is a protein or nucleic acid atom. The second column, 1, is the id of the atom. The third column indicates that it is an N atom. The fourth column, N, is the label of the atom. The fifth column is used to label the isomers. The sixth column, SER, indicates that the atom belongs to a serine residue. The seventh column, A, indicates that the atom belongs to the A chain, the eighth column indicates that the chain number is 1, the ninth column indicates that the residue number is 1. columns 11, 12, and 13 correspond to the x, y, and z coordinates of the atom, respectively. Column 14, 1.00, indicates that the molecule's conformational occupancy in the crystal is 100%. Column 15, 40.61, indicates that the electron density drags the tail (the larger the value, the more violent the atomic motion). Columns 17 and 18 indicate that the residue is 1 and serine. Column 19 indicates that the chain is the A-chain. Column 20 indicates that the atom is a member of an A-chain. Column 21 indicates the model number.\n",
    "\n",
    "#### 5ZMD\n",
    "```\n",
    "ATOM   1     N  N     . GLU A 1 1   ? -10.252 6.628   -28.195 1.00 27.74  ? 37   GLU A N     1 \n",
    "ATOM   2     C  CA    . GLU A 1 1   ? -11.347 7.590   -28.099 1.00 34.30  ? 37   GLU A CA    1 \n",
    "ATOM   3     C  C     . GLU A 1 1   ? -12.638 6.888   -27.685 1.00 41.38  ? 37   GLU A C     1 \n",
    "ATOM   4     O  O     . GLU A 1 1   ? -13.436 7.435   -26.925 1.00 47.19  ? 37   GLU A O     1 \n",
    "```\n",
    "> These 4 rows corresponds to the first 4 atoms. The first atom is N, the second and third atoms are the C, and the forth atom is not recognized. All of them are belong to the Glu and A chain. The columns 11, 12, and 13 correspond to the x, y, and z coordinates of the atoms.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b969949",
   "metadata": {},
   "source": [
    "### Chains And Models\n",
    "#### 7PFL\n",
    "```\n",
    "ATOM   2461 C  CE1 . PHE A 1 305 ? 8.670  -18.664 -8.989  1.00 52.31  ? 305 PHE A CE1 1 \n",
    "ATOM   2462 C  CE2 . PHE A 1 305 ? 6.407  -18.054 -8.529  1.00 49.56  ? 305 PHE A CE2 1 \n",
    "ATOM   2463 C  CZ  . PHE A 1 305 ? 7.708  -17.692 -8.813  1.00 52.27  ? 305 PHE A CZ  1 \n",
    "HETATM 2464 NA NA  . NA  B 2 .   ? 9.522  -15.992 20.820  1.00 22.42  ? 401 NA  A NA  1 \n",
    "HETATM 2465 O  O   . HOH C 3 .   ? 22.181 -12.394 27.888  1.00 55.97  ? 501 HOH A O   1 \n",
    "HETATM 2466 O  O   . HOH C 3 .   ? 9.530  -38.557 14.369  1.00 51.61  ? 502 HOH A O   1 \n",
    "```\n",
    "> The six rows of atomic records above show the transition from the A chain (chain 1) to the B chain (chain 2), and the C chain (chain 3).The HETATM in the B and C chains indicates that the atoms are located on small molecules rather than on proteins or nucleic acids. There is only one Na atom on the B chain. Only the A chain is a polypeptide chain, with a total of 2463 atoms recorded.\n",
    "\n",
    "```\n",
    "HETATM 2747 O  O   . HOH C 3 .   ? 24.466 6.386   -19.399 1.00 48.69  ? 783 HOH A O   1 \n",
    "```\n",
    "> Above is the last row of atoms. Columns 2 and 8 show that there are 2,747 atoms in the record, 3 chains, and the last column shows that there is only one model.\n",
    "#### 5ZMD\n",
    "```\n",
    "ATOM   3281  O  OE1   . GLN A 1 463 ? 40.361  38.483  -38.349 1.00 46.95  ? 499  GLN A OE1   1 \n",
    "ATOM   3282  N  NE2   . GLN A 1 463 ? 41.062  36.636  -39.359 1.00 59.09  ? 499  GLN A NE2   1 \n",
    "ATOM   3283  P  P     . DT  B 2 1   ? 26.172  17.712  -30.071 1.00 42.85  ? 2    DT  B P     1 \n",
    "ATOM   3284  O  OP1   . DT  B 2 1   ? 26.730  16.605  -30.881 1.00 38.76  ? 2    DT  B OP1   1 \n",
    "...\n",
    "\n",
    "ATOM   3464  N  N3    . DG  B 2 9   ? -9.020  34.701  -31.719 1.00 17.22  ? 10   DG  B N3    1 \n",
    "ATOM   3465  C  C4    . DG  B 2 9   ? -8.311  35.466  -30.853 1.00 17.07  ? 10   DG  B C4    1 \n",
    "ATOM   3466  N  N     . GLU C 1 1   ? 25.047  40.785  -43.084 1.00 41.34  ? 37   GLU C N     1 \n",
    "ATOM   3467  C  CA    . GLU C 1 1   ? 24.324  40.965  -44.331 1.00 39.38  ? 37   GLU C CA    1 \n",
    "...\n",
    "\n",
    "ATOM   6694  O  OE1   . GLN C 1 463 ? -13.288 -13.523 -30.438 1.00 47.14  ? 499  GLN C OE1   1 \n",
    "ATOM   6695  N  NE2   . GLN C 1 463 ? -12.898 -13.700 -28.245 1.00 49.46  ? 499  GLN C NE2   1 \n",
    "ATOM   6696  P  P     . DT  D 2 1   ? 13.635  4.219   -42.912 1.00 60.88  ? 2    DT  D P     1 \n",
    "ATOM   6697  O  OP1   . DT  D 2 1   ? 12.593  4.366   -41.871 1.00 45.71  ? 2    DT  D OP1   1 \n",
    "...\n",
    "\n",
    "ATOM   6877  N  N3    . DG  D 2 9   ? -4.658  24.020  -37.797 1.00 19.14  ? 10   DG  D N3    1 \n",
    "ATOM   6878  C  C4    . DG  D 2 9   ? -5.637  24.433  -36.959 1.00 20.02  ? 10   DG  D C4    1 \n",
    "ATOM   6879  N  N     . GLU E 1 1   ? 37.219  -41.991 -40.266 1.00 41.10  ? 37   GLU E N     1 \n",
    "ATOM   6880  C  CA    . GLU E 1 1   ? 35.875  -41.701 -40.737 1.00 47.94  ? 37   GLU E CA    1 \n",
    "...\n",
    "\n",
    "ATOM   10105 C  C     . GLY E 1 462 ? 70.569  11.245  -28.016 1.00 54.38  ? 498  GLY E C     1 \n",
    "ATOM   10106 O  O     . GLY E 1 462 ? 71.598  11.830  -27.678 1.00 57.31  ? 498  GLY E O     1 \n",
    "ATOM   10107 P  P     . DT  F 2 1   ? 48.052  -4.381  -38.644 1.00 47.26  ? 2    DT  F P     1 \n",
    "ATOM   10108 O  OP1   . DT  F 2 1   ? 47.228  -4.831  -37.501 1.00 37.95  ? 2    DT  F OP1   1 \n",
    "...\n",
    "\n",
    "ATOM   10288 N  N3    . DG  F 2 9   ? 66.671  -23.837 -33.848 1.00 22.79  ? 10   DG  F N3    1 \n",
    "ATOM   10289 C  C4    . DG  F 2 9   ? 67.716  -24.316 -33.131 1.00 27.60  ? 10   DG  F C4    1 \n",
    "ATOM   10290 N  N     . GLU G 1 1   ? 71.132  -6.515  -25.000 1.00 35.88  ? 37   GLU G N     1 \n",
    "ATOM   10291 C  CA    . GLU G 1 1   ? 71.975  -7.170  -24.005 1.00 40.05  ? 37   GLU G CA    1 \n",
    "...\n",
    "\n",
    "ATOM   13582 O  OE1   . GLN G 1 463 ? 16.277  -42.932 -39.124 1.00 51.97  ? 499  GLN G OE1   1 \n",
    "ATOM   13583 N  NE2   . GLN G 1 463 ? 15.454  -41.291 -40.418 1.00 53.81  ? 499  GLN G NE2   1 \n",
    "ATOM   13584 P  P     . DT  H 2 1   ? 35.079  -18.687 -24.440 1.00 50.93  ? 2    DT  H P     1 \n",
    "ATOM   13585 O  OP1   . DT  H 2 1   ? 33.937  -18.119 -25.191 1.00 35.90  ? 2    DT  H OP1   1 \n",
    "...\n",
    "\n",
    "ATOM   13765 N  N3    . DG  H 2 9   ? 70.279  -34.566 -27.822 1.00 15.11  ? 10   DG  H N3    1 \n",
    "ATOM   13766 C  C4    . DG  H 2 9   ? 69.582  -35.408 -27.021 1.00 11.04  ? 10   DG  H C4    1 \n",
    "HETATM 13767 MN MN    . MN  I 3 .   ? 7.558   13.260  -22.642 1.00 12.01  ? 1001 MN  A MN    1 \n",
    "HETATM 13768 C  C1    . OGA J 4 .   ? 6.705   14.838  -20.412 1.00 16.00  ? 1002 OGA A C1    1 \n",
    "HETATM 13769 C  C2    . OGA J 4 .   ? 5.622   13.931  -20.772 1.00 15.95  ? 1002 OGA A C2    1 \n",
    "...\n",
    "\n",
    "HETATM 13776 N  N1    . OGA J 4 .   ? 4.376   13.859  -19.899 1.00 21.56  ? 1002 OGA A N1    1 \n",
    "HETATM 13777 O  O4    . OGA J 4 .   ? 1.675   11.458  -19.384 1.00 32.98  ? 1002 OGA A O4    1 \n",
    "HETATM 13778 MN MN    . MN  K 3 .   ? 17.259  22.811  -49.946 1.00 17.35  ? 1001 MN  C MN    1 \n",
    "HETATM 13779 C  C1    . OGA L 4 .   ? 16.055  24.608  -52.289 1.00 18.27  ? 1002 OGA C C1    1 \n",
    "HETATM 13780 C  C2    . OGA L 4 .   ? 17.016  25.645  -51.934 1.00 18.30  ? 1002 OGA C C2    1 \n",
    "...\n",
    "\n",
    "HETATM 13787 N  N1    . OGA L 4 .   ? 17.123  26.900  -52.790 1.00 12.23  ? 1002 OGA C N1    1 \n",
    "HETATM 13788 O  O4    . OGA L 4 .   ? 17.851  28.930  -54.469 1.00 14.16  ? 1002 OGA C O4    1 \n",
    "HETATM 13789 MN MN    . MN  M 3 .   ? 43.689  -22.947 -46.346 1.00 20.15  ? 1001 MN  E MN    1 \n",
    "HETATM 13790 C  C1    . OGA N 4 .   ? 45.681  -23.941 -48.384 1.00 24.34  ? 1002 OGA E C1    1 \n",
    "HETATM 13791 C  C2    . OGA N 4 .   ? 44.715  -24.993 -48.085 1.00 18.40  ? 1002 OGA E C2    1 \n",
    "...\n",
    "\n",
    "HETATM 13798 N  N1    . OGA N 4 .   ? 44.499  -26.124 -49.080 1.00 20.72  ? 1002 OGA E N1    1 \n",
    "HETATM 13799 O  O4    . OGA N 4 .   ? 43.889  -28.270 -50.654 1.00 14.73  ? 1002 OGA E O4    1 \n",
    "HETATM 13800 MN MN    . MN  O 3 .   ? 53.341  -13.298 -19.151 1.00 13.77  ? 1001 MN  G MN    1 \n",
    "HETATM 13801 C  C1    . OGA P 4 .   ? 55.329  -14.818 -16.864 1.00 17.54  ? 1002 OGA G C1    1 \n",
    "HETATM 13802 C  C2    . OGA P 4 .   ? 56.409  -13.888 -17.166 1.00 11.52  ? 1002 OGA G C2    1 \n",
    "...\n",
    "\n",
    "HETATM 13809 N  N1    . OGA P 4 .   ? 57.497  -13.631 -16.132 1.00 12.01  ? 1002 OGA G N1    1 \n",
    "HETATM 13810 O  O4    . OGA P 4 .   ? 60.364  -11.454 -15.766 1.00 30.35  ? 1002 OGA G O4    1 \n",
    "HETATM 13811 O  O     . HOH Q 5 .   ? 19.417  4.949   -7.206  1.00 41.58  ? 1101 HOH A O     1 \n",
    "HETATM 13812 O  O     . HOH Q 5 .   ? -2.066  34.376  -22.334 1.00 17.15  ? 1102 HOH A O     1 \n",
    "...\n",
    "\n",
    "HETATM 13819 O  O     . HOH Q 5 .   ? -6.597  6.694   -5.764  1.00 18.12  ? 1109 HOH A O     1 \n",
    "HETATM 13820 O  O     . HOH Q 5 .   ? -3.821  27.200  -25.901 1.00 12.73  ? 1110 HOH A O     1 \n",
    "HETATM 13821 O  O     . HOH R 5 .   ? 7.549   -6.969  -50.541 1.00 33.99  ? 1101 HOH C O     1 \n",
    "HETATM 13822 O  O     . HOH R 5 .   ? -6.047  34.496  -42.571 1.00 26.42  ? 1102 HOH C O     1 \n",
    "HETATM 13823 O  O     . HOH R 5 .   ? 6.203   35.175  -54.675 1.00 17.33  ? 1103 HOH C O     1 \n",
    "HETATM 13824 O  O     . HOH S 5 .   ? -1.446  32.506  -36.085 1.00 17.10  ? 101  HOH D O     1 \n",
    "HETATM 13825 O  O     . HOH S 5 .   ? -5.768  34.513  -37.580 1.00 18.86  ? 102  HOH D O     1 \n",
    "HETATM 13826 O  O     . HOH T 5 .   ? 36.869  -37.338 -63.112 1.00 15.89  ? 1101 HOH E O     1 \n",
    "HETATM 13827 O  O     . HOH U 5 .   ? 58.445  -20.199 -31.565 1.00 15.06  ? 1101 HOH G O     1 \n",
    "HETATM 13828 O  O     . HOH U 5 .   ? 48.053  -35.199 -29.274 1.00 23.13  ? 1102 HOH G O     1 \n",
    "```\n",
    "> These rows record the transition from the A chain to the U chain. In the '_atom_site.label_comp_id' column, DT refers to the Deoxythymidine, DT refers to the Deoxyguanosine, MN refers to the Manganese ion, OGA refers to the N-oxalylglycine, HOH refers to the water, and the others refer to the amino acid. Thus, we can know that chain A, C, E and G are peptide chains, chain B, D, F, and H are DNA chains, chain I, K, M and O are Manganese ions, chain J, L, N and P are N-oxalylglycine, and chain Q, R, S, T and U are water."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fe8829bf",
   "metadata": {},
   "source": [
    "### Conformation\n",
    "#### 7PFL\n",
    "```\n",
    "ATOM   1299 C  CE1 . HIS A 1 164 ? 16.200 -17.519 16.231  1.00 18.53  ? 164 HIS A CE1 1 \n",
    "ATOM   1300 N  NE2 . HIS A 1 164 ? 15.912 -17.316 14.958  1.00 21.49  ? 164 HIS A NE2 1 \n",
    "ATOM   1301 N  N   A MET A 1 165 ? 12.337 -12.037 17.064  0.55 18.89  ? 165 MET A N   1 \n",
    "ATOM   1302 N  N   B MET A 1 165 ? 12.357 -12.029 17.055  0.45 18.91  ? 165 MET A N   1 \n",
    "ATOM   1303 C  CA  A MET A 1 165 ? 11.981 -11.059 18.075  0.55 24.13  ? 165 MET A CA  1 \n",
    "ATOM   1304 C  CA  B MET A 1 165 ? 12.187 -10.937 18.007  0.45 23.99  ? 165 MET A CA  1 \n",
    "```\n",
    "> Column 5 refers to the alternative conformations, while column 14 is the occupancy of the conformations present, indicating the fraction of atoms with each conformation. The third and fourth rows above show that atoms 1301 and 1302 are the A and B conformations of the N atom of the MET amino acid residue, with occupancies of 55% and 45%, respectively.\n",
    "#### 5ZMD\n",
    "> It is surprising that in the 5ZMD record, every atom lacks alternative conformations, suggesting that atoms without such variations are quite common."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "681d0434",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "### Biological Assembly\n",
    "#### 7PFL\n",
    "```\n",
    "_pdbx_struct_assembly.id                   1 \n",
    "_pdbx_struct_assembly.details              author_and_software_defined_assembly \n",
    "_pdbx_struct_assembly.method_details       PISA \n",
    "_pdbx_struct_assembly.oligomeric_details   dimeric \n",
    "_pdbx_struct_assembly.oligomeric_count     2 \n",
    "```\n",
    "> In _pdbx_struct_assembly: line 1 (id) identifies the assembly as 1. Line 2 (details) indicates that the assembly was confirmed by the authors and the computational software. Line 3 (method_details) indicates that the software used was PISA. Line 4 (oligomeric_details) indicates that the assembly is dimeric, and line 5 (oligomeric_count) confirms that there are 2 subunits in the assembly.\n",
    "\n",
    "```\n",
    "_pdbx_struct_assembly_gen.assembly_id       1 \n",
    "_pdbx_struct_assembly_gen.oper_expression   1,2 \n",
    "_pdbx_struct_assembly_gen.asym_id_list      A,B,C \n",
    "```\n",
    "> In _pdbx_struct_assembly_gen: line 1 (id) again states that the assembly is 1. Line 2 (oper_expression) is the id for the operator, and line 3 has the IDs of the asymmetric units of the assembly as A, B, and C, respectively.\n",
    "\n",
    "```\n",
    "loop_\n",
    "_pdbx_struct_oper_list.id \n",
    "_pdbx_struct_oper_list.type \n",
    "_pdbx_struct_oper_list.name \n",
    "_pdbx_struct_oper_list.symmetry_operation \n",
    "_pdbx_struct_oper_list.matrix[1][1] \n",
    "_pdbx_struct_oper_list.matrix[1][2] \n",
    "_pdbx_struct_oper_list.matrix[1][3] \n",
    "_pdbx_struct_oper_list.vector[1] \n",
    "_pdbx_struct_oper_list.matrix[2][1] \n",
    "_pdbx_struct_oper_list.matrix[2][2] \n",
    "_pdbx_struct_oper_list.matrix[2][3] \n",
    "_pdbx_struct_oper_list.vector[2] \n",
    "_pdbx_struct_oper_list.matrix[3][1] \n",
    "_pdbx_struct_oper_list.matrix[3][2] \n",
    "_pdbx_struct_oper_list.matrix[3][3] \n",
    "_pdbx_struct_oper_list.vector[3] \n",
    "1 'identity operation'         1_555 x,y,z   1.0000000000  0.0000000000 0.0000000000 0.0000000000 0.0000000000 1.0000000000 \n",
    "0.0000000000 0.0000000000 0.0000000000 0.0000000000 1.0000000000  0.0000000000 \n",
    "2 'crystal symmetry operation' 2_555 -x,y,-z -1.0000000000 0.0000000000 0.0000000000 0.0000000000 0.0000000000 1.0000000000 \n",
    "0.0000000000 0.0000000000 0.0000000000 0.0000000000 -1.0000000000 0.0000000000 \n",
    "```\n",
    "> In pdbx_struct_oper_list: this section indicates the operations needed to generate the biological assemblies. Column 1 (id) is the id of the operator. It was mentioned above that there are 1 and 2 operators. Column 2 (type) indicates the type of operators. 'identity operation' indicates that the unit does not perform the operation, and 'crystal symmetry operation' indicates that the unit performs the symmetry operation. The column 3 (name) indicates the type of operator, which indicates the transformation method of the operator. 5 indicates no translation, so here 1_555 and 2_555 indicate that operator 1 and 2 have no translations in the x, y, and z directions. The next three columns indicate that operator 2 does symmetry in the yz-plane and xy-plane, respectively, and thus the coordinates are -x, y, and -z.\n",
    "#### 5ZMD\n",
    "```\n",
    "_pdbx_struct_assembly.id \n",
    "_pdbx_struct_assembly.details \n",
    "_pdbx_struct_assembly.method_details \n",
    "_pdbx_struct_assembly.oligomeric_details \n",
    "_pdbx_struct_assembly.oligomeric_count \n",
    "1 author_defined_assembly ? tetrameric 4 \n",
    "2 author_defined_assembly ? tetrameric 4 \n",
    "```\n",
    "> This section indicates that both assembly 1 and assembly 2 are tetrameric, which is confirmed only by the author.\n",
    "\n",
    "```\n",
    "_pdbx_struct_assembly_gen.assembly_id \n",
    "_pdbx_struct_assembly_gen.oper_expression \n",
    "_pdbx_struct_assembly_gen.asym_id_list \n",
    "1 1 A,B,C,D,I,J,K,L,Q,R,S \n",
    "2 1 E,F,G,H,M,N,O,P,T,U   \n",
    "```\n",
    "> The chain A,B,C,D,I,J,K,L,Q,R,S (2 peptide chains + 2 DNA chains + 2 Manganese ions + 2 N-oxalylglycine + water) belong to the assembly 1, and the chain E,F,G,H,M,N,O,P,T,U (2 peptide chains + 2 DNA chains + 2 Manganese ions + 2 N-oxalylglycine + water) belong to the assembly 2. Both assembly conduct the operation 1.\n",
    "```\n",
    "_pdbx_struct_oper_list.id                   1 \n",
    "_pdbx_struct_oper_list.type                 'identity operation' \n",
    "_pdbx_struct_oper_list.name                 1_555 \n",
    "_pdbx_struct_oper_list.symmetry_operation   x,y,z \n",
    "_pdbx_struct_oper_list.matrix[1][1]         1.0000000000 \n",
    "_pdbx_struct_oper_list.matrix[1][2]         0.0000000000 \n",
    "_pdbx_struct_oper_list.matrix[1][3]         0.0000000000 \n",
    "_pdbx_struct_oper_list.vector[1]            0.0000000000 \n",
    "_pdbx_struct_oper_list.matrix[2][1]         0.0000000000 \n",
    "_pdbx_struct_oper_list.matrix[2][2]         1.0000000000 \n",
    "_pdbx_struct_oper_list.matrix[2][3]         0.0000000000 \n",
    "_pdbx_struct_oper_list.vector[2]            0.0000000000 \n",
    "_pdbx_struct_oper_list.matrix[3][1]         0.0000000000 \n",
    "_pdbx_struct_oper_list.matrix[3][2]         0.0000000000 \n",
    "_pdbx_struct_oper_list.matrix[3][3]         1.0000000000 \n",
    "_pdbx_struct_oper_list.vector[3]            0.0000000000 \n",
    "```\n",
    "> This section indicates how to conduct the operation 1. 'Identity operation' and 1_555 mean that there is no movement in the operation 1."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9bde2748",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
