{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0b31f192",
   "metadata": {},
   "source": [
    "# M01(VL01) Daiyun Dong"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "601283bb",
   "metadata": {},
   "source": [
    "## 3SOV\n",
    "\n",
    "From the PDB/MMCIF file, it is evident that the file contains two polypeptide chains. Chain A, which is our focus, is a β-helix with a MW of 35,685.926 Daltons, and Chain B is Sclerostin with a MW of 764.917 Daltons. The corresponding image is shown below, presenting the two polypeptide chains, with the larger one being the protein of our interest.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/kq08r8.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/ukjun1.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/5ug3mp.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "The β-helix resembles a propeller. In this protein, there are six blades arranged around a central axis, with each blade comprising interactions of four antiparallel β-strands.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/9udyv8.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/2g66f1.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/er9hg8.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "We need to note that the presence of four antiparallel β-strands interacting within each blade does not imply that each blade contains four complete, sequentially continuous β-sheets. Blades 1 and 2 share one β-strand; at the same time, blade 1 includes three β-strands at the N-terminus (colored blue in PyMOL) and one β-strand at the C-terminus (colored orange in PyMOL). The stability of the protein depends on the interactions among molecules and secondary structures in space, rather than the stability of the sequence in the primary structure. Therefore, it is reasonable for two blades to share one β-strand and for all β-strands within a blade to be non-sequentially continuous.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/6tvtny.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/jezocq.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/gc5myb.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "。\n",
    "In antiparallel β-sheets, the atoms of each strand can form hydrogen bonds with the atoms of adjacent strands; and the antiparallel arrangement allows the hydrogen bonds to align almost linearly, resulting in greater stability and a more compact spatial arrangement. Taking blade 6 as an example, the distance between two adjacent β-strands is about 5 Å, and they are arranged in an antiparallel manner.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/jsp6h5.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/97lzdm.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/vrtj8k.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "There are numerous parallel hydrogen bonds between adjacent strands, maintaining the stability of the structure.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/b7454y.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/evuld7.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/5pfs1u.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "By calculating the electrostatic potential, it was found that the center surrounded by the six blades exhibits a strong negative charge. This implies that the central area might be an active site or ligand-binding site, capable of attracting positively charged substrates or ligands, stabilizing the substrate transition states, etc\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/b9jw9i.png\" width=\"250\" height=\"auto\" align=\"middle\" /> | \n",
    "| :----: | \n",
    "|Pymol|\n",
    "\n",
    "\n",
    "\n",
    "We have noticed that Sclerostin is located at the central position of the β-propeller, potentially acting as a substrate or ligand of the β-propeller, closely related to its function. Upon examining the hydrogen bonds, it was found that there are partial hydrogen bonds between Sclerostin and the β-propeller.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/6bwedk.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/q2vmv6.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/oqk7se.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "By setting hydrophobic amino acids (Gly, Ala, Val, Leu, Ile, Pro, Phe, Met, Trp) to yellow and others to blue in PyMOL and Chimera, it was observed that the distribution of hydrophobic and hydrophilic amino acids is relatively even, leading to the conclusion that there might not be regions within the β-propeller with strong hydrophobicity. Sclerostin, located at the center of the β-propeller, exhibits strong hydrophilicity, suggesting that the center of the β-propeller might be more inclined towards hydrophilicity. (In VMD, hydrophobic amino acids: nonpolar amino acids are white. Hydrophilic amino acids: polar amino acids with a positive charge are blue, polar amino acids with a negative charge are red, and polar amino acids without a charge are green.)\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/l3435t.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/5t4ifc.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/vilgxy.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9cd7d8c1",
   "metadata": {},
   "source": [
    "## 3TDM\n",
    "\n",
    "From the PDB/MMCIF file, it is apparent that there are four polypeptide chains in the file, together constituting the protein of our interest, a two-fold symmetric TIM-barrel protein with a MW of 13,581.423 Daltons. Additionally, the document also records the protein's ligands -- four phosphate ions. Furthermore, there are two assemblies within this protein, both are dimeric, and the two subunits of the second assembly are symmetric. Observing the image, the two subunits of assembly2 are symmetrical, each resembling half an α/β barrel. Assembly1 is a complete α/β barrel fold.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/f9rf7v.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/9mcyie.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/17kiag.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/gm07gx.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/pbmfs0.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/9l8l5k.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "In the α/β barrel of assembly1, there are approximately 8 α-helices on the exterior and about 8 β-sheets arranged around a central axis. Unlike the β-propeller, in the α/β barrel, α-helices and β-sheets alternate in the protein sequence; and the β-sheets arranged around the central axis are parallel rather than antiparallel.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/cifrxi.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/5jwb5j.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/kj48xh.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "In addition to the α/β barrel of assembly1, there are two β-sheets that do not participate in the structure of the α/β barrel. These two β-sheets tend to form a more stable structure on their own, namely, an antiparallel arrangement.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/psndk0.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/wqnx1k.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/ht6dnd.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "We need to note that assembly2 is composed of two halves of an α/β barrel. Since the symmetry of each subunit is lower than that of a complete α/β barrel, to maintain structural stability, the subunits in assembly2 may not fully exhibit the characteristics of an α/β barrel. For example, both subunits each have 6 β-sheets, with the first to third β-sheets arranged in an antiparallel manner and the third to sixth β-sheets in a parallel arrangement. However, they both follow the pattern of α-helices and β-sheets alternating in the protein sequence, with α-helices on the exterior and β-sheets on the interior.\n",
    "\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/b9grbv.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/kt2tdx.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/y1oeo4.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "In the complete α/β barrel of assembly1, the β-strands are arranged in a parallel manner to form the central core of the β-barrel. On the exterior of the β-strands, an α-helix is connected, alternating with the β-strands to form the barrel's outer shell. However, the distance between β-strands in the α/β barrel is about 5 Å, similar to that in the β-propeller.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/8fgyv2.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/i3fwb7.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/5ltq4w.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "There are numerous parallel hydrogen bonds between adjacent β-strands, maintaining the stability of the structure. Additionally, there are many hydrogen bonds within the α-helix and at the junction between the α-helix and β-strands. However, surprisingly, there are almost no hydrogen bonds between the inner β-strands and the outer α-helices in the parallel direction, possibly due to the greater distance.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/li52pp.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/wujmnq.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/8jzjy8.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "By calculating the electrostatic potential, it was found that the center of the α/β barrel exhibits a strong negative charge. This implies that the central area might be an active site or ligand-binding site, capable of attracting positively charged substrates or ligands, stabilizing the substrate transition states, etc. The phosphate ion is located at the center of the α/β barrel, possibly acting as a ligand of the α/β barrel. Considering the negative charge of the phosphate ion, the center of the α/β barrel might exhibit a significant negative charge due to the presence of the phosphate ion.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/qv7v8d.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  \n",
    "| :----: | \n",
    "|Pymol|\n",
    "\n",
    "\n",
    "\n",
    "Observing the hydrogen bonds, it was found that there are numerous hydrogen bonds between the phosphate ion and the α/β barrel, further verifying the interaction between the α/β barrel and the phosphate ion.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/3ohdbp.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/sfhro0.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/p2bbb2.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "By coloring hydrophobic amino acids (Gly, Ala, Val, Leu, Ile, Pro, Phe, Met, Trp) yellow and others blue, it was observed that hydrophobic amino acids are more prevalent on the inside of the α/β barrel. Therefore, it is believed that the center of the α/β barrel is hydrophobic and the exterior is hydrophilic, suggesting that the substrates interacting with it are more likely to be hydrophobic. (In VMD, hydrophobic amino acids: nonpolar amino acids are white. Hydrophilic amino acids: polar amino acids with a positive charge are blue, polar amino acids with a negative charge are red, polar amino acids without a charge are green.)\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/st46w2.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/hcxypd.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/9tto56.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "74e8460b",
   "metadata": {},
   "source": [
    "## 5ZMD\n",
    "\n",
    "\n",
    "From the PDB/MMCIF file, it is evident that the record consists of four polypeptide chains forming the FTO protein with a MW of 53,312.223 Daltons. Additionally, it contains four DNA chains, four manganese ions, and four N-oxalylglycine molecules. In this protein, there are two assemblies, each comprising two polypeptide chains, two DNA chains, two manganese ions, and two N-oxalylglycine molecules. Observing the protein structure, it is found that the secondary and tertiary structures of the four subunits, each with one polypeptide chain and one DNA chain, are also very similar. Conducting blastP comparisons of the first polypeptide chain (chain A) with the second (chain C), third (chain E), and fourth (chain G) polypeptide chains shows that the sequences of the four polypeptide chains are identical. Given that the spatial environment of the four polypeptide chains is also essentially the same, we can infer that their secondary and tertiary structures are fundamentally identical. Therefore, in our analysis, we can focus on the features of a single subunit and the whole. When analyzing a subunit, our primary focus is on the combination of polypeptide chain A, DNA chain B, manganese ion (chain I), and N-oxalylglycine molecule (chain J).\n",
    "\n",
    " <img src=\"https://p.ipic.vip/sl8c47.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    " <img src=\"https://p.ipic.vip/sy0k62.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    " <img src=\"https://p.ipic.vip/hei7kv.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/t92a3b.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/j3ur5t.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/7gs8x8.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "In a subunit, it is mainly divided into two domains. Referring to the study by Han et al. (2010) on the structure of the FTO protein, residues 32-326 constitute the N-terminal domain (NTD), and residues 327-498 make up the C-terminal domain (CTD). In PyMOL, the NTD is represented by the deep blue domain on the right, and the CTD is the light blue domain on the left. Both manganese ions and N-oxalylglycine are located at the center of the NTD, while the DNA chain is situated between the NTD and CTD. In PyMOL, manganese ions are colored green, N-oxalylglycine is red, and the DNA chain is yellow.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/bkva8m.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/pss7z9.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/008b2o.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "Inside the NTD, there are two peptide planes made of antiparallel β-sheets that envelop the manganese ions and N-oxalylglycine. The peptide plane closer to the inside consists of 4 β-sheets, while the outer β-sheet plane comprises 6 β-sheets. On the outermost part, there are two larger α-helices and several smaller α-helices. Referring to the report by Han et al. (2010) on the structure of the FTO protein, the core structure of the NTD belongs to the jelly-roll motif model.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/1k265b.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/fsdkmv.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/30k2t4.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "The CTD (C-terminal domain) is primarily composed of four larger α-helices and two smaller α-helices, connected to the NTD (N-terminal domain) and helps stabilize the NTD.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/xqvi9i.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/ui9d7h.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/owhyzr.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "Measuring the distance between the DNA chain and the CTD, it was found that the closest distance is greater than 10 Å. This result indicates that the DNA chain and the CTD are relatively far apart, suggesting that the CTD may not directly interact with the DNA chain.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/ipks3u.png\" width=\"250\" height=\"auto\" align=\"middle\" /> | <img src=\"https://p.ipic.vip/qjrjy5.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: |\n",
    "|Pymol|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "Within the NTD, there are numerous hydrogen bonds between antiparallel β-sheets, within α-helices, and between antiparallel β-sheets and α-helices, which serve to maintain structural stability. Additionally, there are many hydrogen bonds between β-sheets and manganese ions, as well as N-oxalylglycine, which help stabilize the ligands.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/t4iov2.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/sftuc6.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/2qv2nc.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "Within the CTD, there are numerous hydrogen bonds within the α-helices; in addition, there are many hydrogen bonds between the three α-helices, indicating that they form a stable triple helix bundle.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/41nv2m.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/70z8p2.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/kacd93.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "Hydrogen bonds also exist between the two subunits within the same assembly. These hydrogen bonds link the two subunits together, causing the two DNA chains to be sandwiched between them, making it difficult for them to detach.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/hd82op.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/vf5xf0.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/qz7t7o.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|\n",
    "\n",
    "\n",
    "\n",
    "By calculating the electrostatic potential, it was found that the area surrounding the DNA chains exhibits a negative charge. DNA strands typically display a negative charge due to their phosphate backbone. Therefore, the negative charge at this catalytic center may be attributed to the presence of DNA strands.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/xon9on.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |\n",
    "| :----: | \n",
    "|Pymol|\n",
    "\n",
    "\n",
    "By coloring hydrophobic amino acids (Gly, Ala, Val, Leu, Ile, Pro, Phe, Met, Trp) yellow and other amino acids blue, it was observed that the distribution of hydrophobic and hydrophilic amino acids is relatively even, leading to the conclusion that there might not be regions within the FTO protein with strong hydrophobicity. Given that the DNA chains at the FTO protein's catalytic center are hydrophilic, it is speculated that the FTO protein's catalytic center may be more inclined towards hydrophilicity. (In VMD, hydrophobic amino acids: nonpolar amino acids are colored white. Hydrophilic amino acids: polar amino acids with a positive charge are colored blue, polar amino acids with a negative charge are colored red, and polar amino acids without a charge are colored green.)\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/dv7yn4.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |  <img src=\"https://p.ipic.vip/ylgh2k.png\" width=\"250\" height=\"auto\" align=\"middle\" />  |  <img src=\"https://p.ipic.vip/ro2uka.png\" width=\"250\" height=\"auto\" align=\"middle\" />   |\n",
    "| :----: | :----: | :----: |\n",
    "|Pymol|VMD|UCSF Chimera|"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7502e161",
   "metadata": {},
   "source": [
    "## Reference:\n",
    "Han, Z., Niu, T., Chang, J., Lei, X., Zhao, M., Wang, Q., Cheng, W., Wang, J., Feng, Y., & Chai, J. (2010). Crystal structure of the FTO protein reveals basis for its substrate specificity. Nature, 464(7292), 1205–1209. https://doi.org/10.1038/nature08921"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "adf832b7",
   "metadata": {},
   "source": [
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
