{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e1018ad1",
   "metadata": {},
   "source": [
    "# M06(VL05) Daiyun Dong"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "27bcd020",
   "metadata": {},
   "source": [
    "## Generate Model\n",
    "\n",
    "\n",
    "From the NCBI website, we can obtain the AAC72735.1 protein sequence as follows:\n",
    "```\n",
    ">AAC72735.1 L-lactate dehydrogenase [Trichomonas vaginalis]\n",
    "MSEAAHVLITGAAGQIGYILSHWIASGELYGDRQVYLHLLDIPPAMNRLTALTMELEDCAFPHLAGFVAT\n",
    "TDPKAAFKDIDCAFLVASMPLKPGQVRADLISSNSVIFKNTGEYLSKWAKPSVKVLVIGNPDNTNCEIAM\n",
    "LHAKNLKPENFSSLSMLDQNRAYYEVASKLGVDVKDVHDIIVWGNHGESMVADLTQATFTKEGKTQKVVD\n",
    "VLDHDYVFDTFFKKIGHRAWDILEHRGFTSAASPTKAAIQHMKAWLFGTAPGEVLSMGIPVPEGNPYGIK\n",
    "PGVVFSFPCNVDKEGKIHVVEGFKVNDWLREKLDFTEKDLFHEKEIALNHLAQ\n",
    "```\n",
    "\n",
    "### SWISS-MODEL\n",
    "Using the SWISS-MODEL server to find protein templates, part of the results obtained are shown in the image:\n",
    " \n",
    " <img src=\"https://p.ipic.vip/3akm86.png\" width=\"600\" height=\"auto\" align=\"middle\" />\n",
    "\n",
    "The details are as follows:\n",
    "\n",
    "- Templates: The PDB codes and names of the template protein structures.\n",
    "- Quaternary Structure: The quaternary structure of the template protein (for example, whether it is a monomer or a dimer).\n",
    "- Sequence Similarity: The similarity of the query sequence to the template sequence.\n",
    "- Coverage: The proportion of the query sequence that covers the template sequence.\n",
    "- GMQE: Global Model Quality Estimation, an index calculated by the SWISS-MODEL system that predicts the accuracy of the model.\n",
    "- QSQE: Quaternary Structure Quality Estimation, quality assessment for the quaternary structure.\n",
    "- Identity: The percentage of identity between the query sequence and the template sequence.\n",
    "- Method: The experimental method used to determine the template structure.\n",
    "- Oligo State: The oligomeric state of the protein, indicating how the protein aggregates, such as monomer or dimer.\n",
    "- Ligands: Small molecule ligands bound to the template protein.\n",
    "\n",
    "\n",
    "From the alignment results, the template 4uum.1.A L-LACTATE DEHYDROGENASE is chosen. This template has an identity of 100.0%, resolution of 1.37A, sequence similarity of 62%, and coverage of 100%, indicating that the template can represent the protein of interest well.\n",
    " \n",
    " <img src=\"https://p.ipic.vip/mr7rv2.png\" width=\"250\" height=\"auto\" align=\"middle\" />\n",
    " \n",
    "The model is built using the selected template, and the model quality is as follows:\n",
    "\n",
    "- QMEANDisCo Local: This graph shows the local quality estimate of the model. The Y-axis values are confidence scores predicting local quality, ranging from 0 to 1. The X-axis displays the residue numbers. Overall, the graph is high and steady, indicating that the predicted quality is good across most regions of the model.\n",
    "- GMQE: 0.98 Global Model Quality Estimation is used to predict the reliability of the model. A value of 0.98 indicates a high-quality prediction.\n",
    "- QMEANDisCo Global: 0.92 ± 0.05 It combines the model's local and global consistency scores. This score is also close to 1, indicating a high model quality.\n",
    "- QMEAN: 1.10 This is a comprehensive model quality score. A value above 1 means the model's quality exceeds the average quality of typical PDB structures.\n",
    "- Cβ: 1.33 The carbon beta deviation score, 1.33 means the positions of Cβ atoms in the model are close to ideal values.\n",
    "- All Atom: 0.99 The all-atom score, 0.99 means all atoms' positions are predicted with high precision.\n",
    "- Solvation: 0.84 The solvation score, 0.84 might indicate some parts of the model might not perform optimally in a solvent environment.\n",
    "- Torsion: 0.57 The torsion angle score is relatively low at 0.57, which could mean some parts of the model might have poor torsion angles.\n",
    "- MolProbity Score: 0.54 The MolProbity score comprehensively assesses the overall quality of the structure; the lower the score, the higher the quality, 0.54 indicates good quality.\n",
    "- Clash Score: 0.10 The atomic clash score, 0.10 means there is little spatial conflict between atoms in the model.\n",
    "- Ramachandran Favored: 98.03% Indicates that 98.03% of amino acid residues are in the favored region of the Ramachandran plot.\n",
    "- Ramachandran Outliers: 0.00% No amino acid residues are in the outlier regions of the Ramachandran plot.\n",
    "- Rotamer Outliers: 0.00% The side-chain rotamer outliers are 0%, which is also a positive structural indicator.\n",
    "- C-Beta Deviations: 2 There are two residues whose Cβ atoms deviate from the expected positions.\n",
    "- Bad Bonds: 0/5310 There are no bonds that do not conform to expectations.\n",
    "- Bad Angles: 43/7210 Out of 7210 angles, only 43 angles do not conform to expectations.\n",
    "- Cis Prolines: 2/32 There are two proline residues in the cis conformation, which may be characteristic of specific protein folding or a potential error in the model.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/3srird.png\" width=\"250\" height=\"auto\" align=\"middle\" /> | <img src=\"https://p.ipic.vip/wsf1np.png\" width=\"250\" height=\"auto\" align=\"middle\" /> | <img src=\"https://p.ipic.vip/n0j2kf.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |\n",
    "| :---: | :---: | :---: |\n",
    "\n",
    "\n",
    "The generated model and alignment information are as follows, with about 16 alpha helices and 27 beta sheets.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/0ksoqo.png\" width=\"250\" height=\"auto\" align=\"middle\" /> | <img src=\"https://p.ipic.vip/fr8b86.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |\n",
    "| :---: | :---: |\n",
    "\n",
    "Using ProSA to analyze the model quality, the Z-Score is displayed as -10.33, indicating a high model quality.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/8bpyx5.png\" width=\"250\" height=\"auto\" align=\"middle\" /> | <img src=\"https://p.ipic.vip/u0hrhi.png\" width=\"250\" height=\"auto\" align=\"middle\" /> | <img src=\"https://p.ipic.vip/bz9mr5.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |\n",
    "| :---: | :---: | :---: |\n",
    "\n",
    "Overall, the model has very high quality scores, showing great credibility. This means the predicted structure is reliable compared to the template structure, and the position prediction of each residue in the model has a high degree of accuracy.\n",
    "\n",
    "\n",
    "### Modeller\n",
    "\n",
    "1b8pA, 1bdmA, 1civA, 5mdhA, 7mdhA, and 1smkA were used as input templates for alignment. Partial results are shown below:\n",
    "```\n",
    "Sequence identity comparison (ID_TABLE):\n",
    "\n",
    "   Diagonal       ... number of residues;\n",
    "   Upper triangle ... number of identical residues;\n",
    "   Lower triangle ... % sequence identity, id/min(length).\n",
    "\n",
    "         1b8pA @11bdmA @11civA @25mdhA @27mdhA @21smkA @2\n",
    "1b8pA @1      327     194     147     151     153      49\n",
    "1bdmA @1       61     318     152     167     155      56\n",
    "1civA @2       45      48     374     139     304      53\n",
    "5mdhA @2       46      53      42     333     139      57\n",
    "7mdhA @2       47      49      87      42     351      48\n",
    "1smkA @2       16      18      17      18      15     313\n",
    "\n",
    "\n",
    "Weighted pair-group average clustering based on a distance matrix:\n",
    "\n",
    "\n",
    "                                           .----------------------- 1b8pA @1.9    39.0000\n",
    "                                           |\n",
    "                                  .-------------------------------- 1bdmA @1.8    50.5000\n",
    "                                  |\n",
    "                              .------------------------------------ 5mdhA @2.4    55.3750\n",
    "                              |\n",
    "                              |                                .--- 1civA @2.8    13.0000\n",
    "                              |                                |\n",
    "        .---------------------------------------------------------- 7mdhA @2.4    83.2500\n",
    "        |\n",
    "      .------------------------------------------------------------ 1smkA @2.5\n",
    "\n",
    "      +----+----+----+----+----+----+----+----+----+----+----+----+\n",
    "    86.0600   73.4150   60.7700   48.1250   35.4800   22.8350   10.1900\n",
    "         79.7375   67.0925   54.4475   41.8025   29.1575   16.5125\n",
    "\n",
    "\n",
    "```\n",
    "\n",
    "The identity between protein sequences was calculated as follows:\n",
    "\n",
    "|       | 1b8pA       | 1bdmA       | 1civA       | 5mdhA       | 7mdhA       | 1smkA       |\n",
    "| :-------: | :------: | :------: | :------: | :------: | :-----: | :------: |\n",
    "| 1b8pA | 100.0%      | 61.01%      | 44.95%      | 46.18%      | 46.79%      | 15.65%      |\n",
    "| 1bdmA | 61.01%      | 100.0%      | 47.80%      | 52.52%      | 48.74%      | 17.89%      |\n",
    "| 1civA | 44.95%      | 47.80%      | 100.0%      | 41.74%      | 86.61%      | 16.93%      |\n",
    "| 5mdhA | 46.18%      | 52.52%      | 41.74%      | 100.0%      | 41.74%      | 18.21%      |\n",
    "| 7mdhA | 46.79%      | 48.74%      | 86.61%      | 41.74%      | 100.0%      | 15.34%      |\n",
    "| 1smkA | 15.65%      | 17.89%      | 16.93%      | 18.21%      | 15.34%      | 100.0%      |\n",
    "\n",
    "1b8pA has a resolution of 1.9 A, 1bdmA has a resolution of 1.8 A, 5mdhA has a resolution of 2.4 A, 1civA has a resolution of 2.8 A, 7mdhA has a resolution of 2.4 A, and 1smkA has a resolution of 2.5 A. Among them, 1bdmA having the lowest resolution.\n",
    "\n",
    "Align the 1bdmA with the interested protein.\n",
    "```\n",
    "Pairwise dynamic programming alignment (ALIGN2D):\n",
    "  Residue-residue metric   : (LIB)/as1.sim.mat\n",
    "  Diagonal                 :          100\n",
    "  Overhang                 :            0\n",
    "  Maximal gap length       :       999999\n",
    "  Local alignment          :            F\n",
    "  MATRIX_OFFSET (local aln):       0.0000\n",
    "  FIX_OFFSETS              :      0.0    -1.0    -2.0    -3.0    -4.0\n",
    "  N_SUBOPT                 :            1\n",
    "  SUBOPT_OFFSET            :       0.0000\n",
    "  Alignment block          :            1\n",
    "  Gap introduction penalty :    -900.0000\n",
    "  Gap extension penalty    :     -50.0000\n",
    "  Gap diagonal penalty     :       0.0000\n",
    "  Structure gap penalties  :    3.500   3.500   3.500   0.200   4.000   6.500   2.000   0.000\n",
    "  Length of alignment      :          336\n",
    "  Score                    :  192686.3594\n",
    "\n",
    "```\n",
    "The parameter standards are interpreted as follows:\n",
    "\n",
    "- Residue-residue metric: The residue-residue similarity matrix used is (LIB)/as1.sim.mat.\n",
    "- Diagonal: The diagonal weight is set to 100.\n",
    "- Overhang: The overhang length is set to 0, which means overhangs at the start and end of the sequence alignment are not permitted.\n",
    "- Maximal gap length: The maximum allowed gap length is set to 999999, essentially permitting gaps of any length.\n",
    "- Local alignment: Set to False, indicating that a global alignment is being performed rather than a local alignment.\n",
    "- MATRIX_OFFSET (local aln): The matrix offset for local alignment is 0.0.\n",
    "- FIX_OFFSETS: The setting for fixed offsets, which is used to adjust the gap penalty in sequence alignment, makes the cost of introducing gaps at different positions in the sequence different.\n",
    "- N_SUBOPT: The number of suboptimizations is 1, indicating that only the optimal solution is considered rather than multiple suboptimal solutions.\n",
    "- SUBOPT_OFFSET: The offset for suboptimization is 0.0.\n",
    "- Alignment block: The number of alignment blocks is 1, meaning that the entire alignment is treated as a single block.\n",
    "- Gap introduction penalty: The penalty for introducing a gap is -900.0.\n",
    "- Gap extension penalty: The penalty for extending a gap is -50.0.\n",
    "- Gap diagonal penalty: The penalty for a diagonal gap is 0.0.\n",
    "- Structure gap penalties: Structure gap penalties, where different values correspond to the degree of penalty for gaps in different structural environments.\n",
    "- Length of alignment: The total length of the alignment is 336 residues.\n",
    "- Score: The total score of the alignment is 192686.3594.\n",
    "\n",
    "\n",
    "The alignment result:\n",
    "```\n",
    "\n",
    " _aln.pos         10        20        30        40        50        60\n",
    "1bdmA     MKAPVRVAVTGAAGQIGYSLLFRIAAGEMLGKDQPVILQLLEIPQAMKALEGVVMELEDCAFPLLAGL \n",
    "TvLDH     MSEAAHVLITGAAGQIGYILSHWIASGELYG-DRQVYLHLLDIPPAMNRLTALTMELEDCAFPHLAGF \n",
    " _consrvd *     *  ********* *   ** **  * *  * * ** ** **  *    ********* ***\n",
    "\n",
    " _aln.p   70        80        90       100       110       120       130\n",
    "1bdmA     EATDDPDVAFKDADYALLVGAAPRL---------QVNGKIFTEQGRALAEVAKKDVKVLVVGNPANTN \n",
    "TvLDH     VATTDPKAAFKDIDCAFLVASMPLKPGQVRADLISSNSVIFKNTGEYLSKWAKPSVKVLVIGNPDNTN \n",
    " _consrvd  ** **  **** * * **   *             *  **   *  *   **  ***** *** ***\n",
    "\n",
    " _aln.pos  140       150       160       170       180       190       200\n",
    "1bdmA     ALIAYKNAPGLNPRNFTAMTRLDHNRAKAQLAKKTGTGVDRIRRMTVWGNHSSIMFPDLFHAEVD--- \n",
    "TvLDH     CEIAMLHAKNLKPENFSSLSMLDQNRAYYEVASKLGVDVKDVHDIIVWGNHGESMVADLTQATFTKEG \n",
    " _consrvd   **   *  * * **     ** ***    * * *  *       *****   *  **  *\n",
    "\n",
    " _aln.pos    210       220       230       240       250       260       270\n",
    "1bdmA     -GRPALELVDMEWYEKVFIPTVAQRGAAIIQARGASSAASAANAAIEHIRDWALGTPEGDWVSMAVPS \n",
    "TvLDH     KTQKVVDVLDHDYVFDTFFKKIGHRAWDILEHRGFTSAASPTKAAIQHMKAWLFGTAPGEVLSMGIPV \n",
    " _consrvd          *       *      *   *   **  ****   *** *   *  **  *   **  *\n",
    "\n",
    " _aln.pos      280       290       300       310       320       330\n",
    "1bdmA     Q--GEYGIPEGIVYSFPVTAK-DGAYRVVEGLEINEFARKRMEITAQELLDEMEQVKAL--GLI \n",
    "TvLDH     PEGNPYGIKPGVVFSFPCNVDKEGKIHVVEGFKVNDWLREKLDFTEKDLFHEKEIALNHLAQGG \n",
    " _consrvd      ***  * * ***      *   ****   *   *     *   *  * *\n",
    "```\n",
    "\n",
    "There are 135 matches, 182 mismatches, and 19 gaps, and the sequence identity is 42.59%. The template 1bdmA was chosen for model construction.\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Part of model-single.log is shown below：\n",
    "```\n",
    ">> Summary of successfully produced models:\n",
    "Filename                          molpdf     DOPE score    GA341 score\n",
    "----------------------------------------------------------------------\n",
    "TvLDH.B99990001.pdb           1763.56104   -38079.76172        1.00000\n",
    "TvLDH.B99990002.pdb           1560.93396   -38515.98047        1.00000\n",
    "TvLDH.B99990003.pdb           1712.44104   -37984.30859        1.00000\n",
    "TvLDH.B99990004.pdb           1720.70801   -37869.91406        1.00000\n",
    "TvLDH.B99990005.pdb           1840.91772   -38052.00781        1.00000\n",
    "```\n",
    "The DOPE (Discrete Optimized Protein Energy) score is an energy scoring function based on statistical mechanics. It calculates the energy score of a model by comparing the distances between atom pairs in the target protein model with the distances between corresponding atom pairs in similar protein structures in the database. A lower DOPE score means that the model's structure is closer to the protein structures in the database, indicating a higher quality model. Since the second model has the lowest DOPE score of -38515.98047, the second model TvLDH.B99990002.pdb is the best model.\n",
    "\n",
    "The constructed model structure is as follows, with approximately 8 alpha helices and 10 beta sheets observed, where 6 beta sheets form antiparallel beta-pleated sheets, and 4 beta sheets form parallel beta-pleated sheets. The two beta-pleated sheets are located on both sides, each surrounded by 4 alpha helices.\n",
    "\n",
    "<img src=\"https://p.ipic.vip/clv6mc.png\" width=\"250\" height=\"auto\" align=\"middle\" />\n",
    "\n",
    "The model quality was analyzed using the ProSA server. The local model quality map shows the energy values for each residue position in the model, where lower values (downward parts of the curve) indicate a higher predicted local model quality, while higher values (upward parts of the curve) indicate potential problem areas. In the overall model quality graph, the Z-score is -8.34. This value is in the low-energy area of the Z-score distribution for known protein structures with the corresponding number of residues, indicating a high-quality model.\n",
    "\n",
    "| <img src=\"https://p.ipic.vip/goxykj.png\" width=\"250\" height=\"auto\" align=\"middle\" /> | <img src=\"https://p.ipic.vip/j45xgo.png\" width=\"250\" height=\"auto\" align=\"middle\" /> | <img src=\"https://p.ipic.vip/812k0i.png\" width=\"250\" height=\"auto\" align=\"middle\" /> |\n",
    "| :---: | :---: | :---: |"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3c896f53",
   "metadata": {},
   "source": [
    "## Models Comparison\n",
    "\n",
    "Align two models with PDB Protein Comparison Tool. The results are shown below：\n",
    "<img src=\"https://p.ipic.vip/uky1f1.png\" width=\"800\" height=\"auto\" align=\"middle\" /> \n",
    "<img src=\"https://p.ipic.vip/zoj86y.png\" width=\"400\" height=\"auto\" align=\"middle\" /> \n",
    "<img src=\"https://p.ipic.vip/p0tnyv.png\" width=\"400\" height=\"auto\" align=\"middle\" /> \n",
    "\n",
    "\n",
    "RMSD is a measure of the deviation in relative positions between two structures. A value of 2.64 indicates that, after spatial alignment, the average deviation between corresponding atoms is about 2.64 angstroms. The TM-score is used to assess the topological similarity of the model, with values typically ranging from 0 to 1. A higher value indicates better similarity of the model to the reference structure; a score of 0.92 means that the model from Modeller is very similar in structural topology to the model from SWISS-MODEL. Since both models are built based on the same sequence, both identity and similarity are 100%. The number of equivalent residues is 332, meaning that in the alignment, 332 pairs of residues are considered equivalent.\n",
    "\n",
    "Moreover, in the above analysis, the Z-score for the model built by MODELLER is -8.34, while the Z-SCORE for the model built by SWISS-MODEL is -10.33, indicating SWISS-MODEL performs better in this case.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bc29fc2b",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "808ec7b1",
   "metadata": {},
   "source": [
    "## Discussion\n",
    "### What criteria did you use to identify the template with the lowest resolution and highest sequence identity? Why? Did your process differ between SWISS-MODEL and MODELLER? Why or why not?\n",
    "\n",
    "When the template with the lowest resolution and highest sequence identity is the same template, it means we have found the best template that not only has good structural information, but also is highly similar to the target sequence on a sequence level; therefore, this template is chosen. When they are not the same template, we first filter out all templates with resolutions less than 2A. This is because structures with resolutions lower than 2Å provide detailed atomic position information and are of higher quality. Then, we choose the template with the highest sequence identity to maximize the model's predictive accuracy.\n",
    "\n",
    "Firstly, SWISS-MODEL is a user-friendly interactive website that allows direct template selection and model building on the website. In contrast, MODELLER uses scripts run on the command line, allowing users to better customize the modeling process by editing the script. Secondly, SWISS-MODEL automatically finds templates with high sequence identity to the target sequence in the database, while MODELLER supports manual template selection. Thirdly, SWISS-MODEL uses QMEAN and Z-score to assess model quality, whereas MODELLER evaluates using the DOPE score and allows manual checking and modification of the model.\n",
    "\n",
    "\n",
    "\n",
    "### Which of the predicted local similarity to target (local model quality: DOPE, Discrete Optimized Protein Energy) and the overall model quality (Z-score) for the generated model provide the most useful information regarding the generated model? Why?\n",
    "\n",
    "The DOPE score is based on statistical mechanics principles. It compares the distances of atoms in the model to those in databases protein structures, and measures the model's energy state by calculating the free energy of each amino acid residue in the model. DOPE score is used to assess the quality of local structures in the protein model, helping to identify the state of specific areas in the model.\n",
    "\n",
    "The Z-score compares the quality of our model to a large number of protein structural models, assessing the overall quality of the model, and helping to evaluate whether the model deviates from the normal range of known protein structures on the whole.\n",
    "\n",
    "These two parameters reflect different aspects of the model. The DOPE score is more helpful in checking specific areas of the model, while the Z-score evaluates the model's overall reliability and whether it has the correct folding structure. Therefore, when building a model, both parameters should be checked. We can first use the Z-score for an overall assessment, and then the DOPE score for detailed analysis of specific areas.\n",
    "\n",
    "\n",
    "\n",
    "### Why would the QMEAN provide us with a statistical determination of how good the generated model is? Specifically, think about what contributes to calculation of the QMEAN.\n",
    "\n",
    "QMEAN uses statistical potentials of mean force to quantify the accuracy of the entire model, as well as modeling errors for each residue and globally. It generates an overall score on how close the model is to known proteins by comparing distances between residue pairs in the model to those in a large database of known structures, distributions of angles in the model to those in known protein structures, and distributions of secondary structures in the model to those in known structures, among other factors. By comparing to structures of measured proteins, QMEAN scores the model, and the closer the score is to the corresponding score of experimentally determined structures, the more likely the model is close to the real protein structure.\n",
    "\n",
    "\n",
    "\n",
    "### What goes into the calculation of the DOPE score? What are some defining secondary and tertiary structural elements in the structure you selected based on DOPE score?\n",
    "\n",
    "The DOPE score calculates the energy of a model based on atomic distances and statistical mechanics. It calculates the distance between all pairs of atoms in the protein model, compares them to distances between similar pairs of atoms in databases, and calculates the pairing energy based on the position and type of residue pairs. Additionally, it also considers effects such as solvent. By calculating the DOPE score, we can identify the prediction situation of specific areas in the model. A low DOPE score indicates that specific areas of the model are stable, while a high DOPE score might suggest that the area is unstable or poorly predicted.\n",
    "\n",
    "In the constructed model structure, it was observed to have approximately 8 α-helices and 10 β-sheets, with 6 β-sheets forming antiparallel β-pleated sheets and 4 β-sheets forming parallel β-pleated sheets. The two β-pleated sheets are located on both sides, each surrounded by 4 α-helices. This structural pattern suggests that the model protein might be an α/β protein, where α-helices and β-sheets alternate, with α-helices surrounding the core β-sheets.\n",
    "\n",
    "\n",
    "### What do the RMSD, Z-score, Identity (%), and Similarity (%) tell you about the agreement between the two structures? Specifically, consider each individual contribution to your understanding of structural agreement.\n",
    "\n",
    "\n",
    "RMSD reflects the degree to which atoms in various parts deviate from average positions, evaluating the consistency between the predictive model structure and experimental structure. A low RMSD value indicates that the two structures are very close, with a high degree of atomic position match; a high RMSD value indicates significant structural differences.\n",
    "\n",
    "Z-score represents the overall model quality, measuring the structure's total energy deviation from the energy distribution. It is used to evaluate the consistency between the predictive model structure and experimental structure. A low Z-score indicates the model's quality is similar to the experimental structure, while a high Z-score indicates the model is abnormal compared to known structures.\n",
    "\n",
    "Identity refers to the percentage of identical residues in the total length of matched sequence alignment between two sequences. A high Identity indicates that the two protein sequences are highly conserved, possibly having similar structures and functions. \n",
    "\n",
    "Similarity refers to the percentage of identical and similar residues in the total number of comparisons. Compared to identity, Similarity considers the physicochemical properties of amino acids, assessing the similarity in structure and function. This is because two sequences, even if not identical, similar amino acids can have similar properties, bringing similar structures and functions.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4766ccd3",
   "metadata": {},
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "473779de",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
