{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d07da2ef",
   "metadata": {},
   "source": [
    "# M06(VL04) Daiyun Dong"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "52569cec",
   "metadata": {},
   "source": [
    "\n",
    "## Part I\n",
    "chainA has 54 residues，UniProt code P00268，and the structure is shown in the figure:\n",
    "|<img src=\"https://www.ebi.ac.uk/thornton-srv/databases/cgi-bin/pdbsum/getimg.pl?source=pdbsum&pdb_code=1b13&file=chainA.png\" width=\"400\" height=\"auto\" align=\"middle\" />| <img src=\"https://www.ebi.ac.uk/thornton-srv/databases/cgi-bin/pdbsum/getimg.pl?source=pdbsum&pdb_code=1b13&file=domA01.gif\" width=\"250\" height=\"auto\" align=\"middle\" />|\n",
    "| :---: | :---: |\n",
    "\n",
    "The structure is documented as follows:\n",
    "<img src=\"https://p.ipic.vip/ntiue6.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    "<img src=\"https://p.ipic.vip/6omwpg.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    "<img src=\"https://p.ipic.vip/cxsdf9.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    "\n",
    "The protein sequence is::\n",
    "```\n",
    ">pdb|1B13|A Chain A, PROTEIN (RUBREDOXIN)\n",
    "MKKYTCTVCAYIYNPEDGDPDNGVNPGTDFKDIPDDWVCPLCGVGKDQFEEVEE\n",
    "```\n",
    "After changing residue A10 to G10, the sequence is:\n",
    "```\n",
    ">pdb|1B13|A Chain A, PROTEIN (RUBREDOXIN)\n",
    "MKKYTCTVCGYIYNPEDGDPDNGVNPGTDFKDIPDDWVCPLCGVGKDQFEEVEE\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f83ebd80",
   "metadata": {},
   "source": [
    "## Part II\n",
    "### PHD\n",
    " -  Alpha helix     (Hh) :     0 is   0.00%\n",
    " -  310  helix       (Gg) :     0 is   0.00%\n",
    " -  Pi helix        (Ii) :     0 is   0.00%\n",
    " -  Beta bridge     (Bb) :     0 is   0.00%\n",
    " -  Extended strand (Ee) :    12 is  22.22%\n",
    " -  Beta turn       (Tt) :     0 is   0.00%\n",
    " -  Bend region     (Ss) :     0 is   0.00%\n",
    " -  Random coil     (Cc) :    42 is  77.78%\n",
    " -  Ambiguous states (?)  :     0 is   0.00%\n",
    " -  Other states         :     0 is   0.00%\n",
    "\n",
    "|<img src=\"https://p.ipic.vip/ljntkd.png\" width=\"500\" height=\"auto\" align=\"middle\" />| <img src=\"https://p.ipic.vip/lyh7hh.png\" width=\"400\" height=\"auto\" align=\"middle\" />|\n",
    "| :---: | :---: |\n",
    "\n",
    "### PSIPRED\n",
    "\n",
    "<img src=\"https://p.ipic.vip/325ecn.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    "\n",
    "\n",
    "### JPred4\n",
    "\n",
    "<img src=\"https://p.ipic.vip/hn7t4l.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    "\n",
    "\n",
    "### RaptorX\n",
    "Unfortunately, due to too many connections, RaptorX cannot work.\n",
    "<img src=\"https://p.ipic.vip/ic0ocy.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    "\n",
    "\n",
    "### PredictProtein 2013\n",
    "\n",
    "\n",
    "<img src=\"https://p.ipic.vip/26e76u.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    "\n",
    "\n",
    "\n",
    "### PEP2D\n",
    "\n",
    "<img src=\"https://p.ipic.vip/6jn373.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    "\n",
    "Cut off 1-4 amino acid.\n",
    "\n",
    "\n",
    "### 2dSS\n",
    "\n",
    "It's used to visualize the secondary structure of multiple sequence alignment or multiple prediction result. \n",
    "\n",
    "<img src=\"https://p.ipic.vip/p133vs.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    "\n",
    "\n",
    "### PROTEUS2\n",
    " \t\n",
    "- Time of Submission: 14:01:37 Mar 04, 2024\n",
    "- Sequence Name: pdb|1B13|A Chain A, PROTEIN (RUBREDOXIN)\n",
    "- Number of residues read in: 54\n",
    "- Sequence does not contain a transmembrane helicle region.\n",
    "- Sequence does not contain a signal peptide.\n",
    "- Number of useable PDB homologs found: 2\n",
    "- 1IRO , e-value = 1.0E-30\n",
    "- 5RXN , e-value = 1.0E-29\n",
    "- Number of sequence alignments used for ab-initio predictions: 49\n",
    "- Overall confidence value: 72.6%\n",
    "- Predicted % Helix content: 0 % (0 residues)\n",
    "- Predicted % Beta sheet content: 26 % (14 residues)\n",
    "- Predicted % Coil content: 74 % (40 residues)\n",
    "- Predicted % Signal peptide content: 0 % (0 residues)\n",
    "- Predicted % membrane content: 0 % (0 residues)\n",
    "\n",
    "<img src=\"https://p.ipic.vip/z5hbqe.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71d20225",
   "metadata": {},
   "source": [
    "## Part III\n",
    "### Observed secondary structure\n",
    "\n",
    "<img src=\"https://p.ipic.vip/ntiue6.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    "<img src=\"https://p.ipic.vip/6omwpg.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    "<img src=\"https://p.ipic.vip/cxsdf9.png\" width=\"900\" height=\"auto\" align=\"middle\" />\n",
    "\n",
    "The protein has a total of 54 residues. The first helix is from 20 to 22, the second helix from 30 to 32, and the third helix from 46 to 48. The first beta strand is from 4 to 6, the second beta strand from 12 to 14, and the third beta strand from 49 to 51. The first turn is from 7 to 9, the second turn from 15 to 17, and the third turn from 40 to 42.\n",
    "\n",
    "### PSIPRED\n",
    "In the PSIPRED prediction data, residues 3-5 are beta strands, residues 12-13 are beta strands, residues 30-32 are helices, residues 46-48 are helices, and residues 50-51 are beta strands. All other residues are predicted as coils.\n",
    "\n",
    "Thus, \n",
    "- Q_helix(observed) =(3+3)/(3+3+3) = 66.7%, Q_strand(observed)=(2+2+2)/(3+3+3) = 66.7%, Q_loop(observed)= 35/(54-3-3-3-3-3-3)=97.2%\n",
    "- Q_helix(predicted) =(3+3)/(3+3) = 100%, Q_strand(observed)=(2+2+2)/(3+2+2)= 85.7%,Q_loop(observed)= 35/(54-3-2-3-3-2)=85.4%\n",
    "- Q3 = [(3+3)+(2+2+2)+35]/54 = 87.0%\n",
    "\n",
    "### PHD\n",
    "In the PHD prediction data, residues 4-7 are extended strands, residues 10-13 are extended strands, residues 30-32 are extended strands, and residue 38 is an extended strand.\n",
    "\n",
    "Thus, \n",
    "- Q_helix(observed) = 0/(3+3+3) = 0%, Q_strand(observed)=5/(3+3+3) = 55.6%, Q_loop(observed)= 32/(54-3-3-3-3-3-3)=88.9%\n",
    "- Q_strand(observed)=5/(4+4+3+1)= 41.7%, Q_loop(observed)= 32/(54-4-4-3-1)=76.2%\n",
    "- Q3 = [5+32]/54 = 68.5%\n",
    "\n",
    "\n",
    "By comparing the Q3 scores, we can conclude that the prediction results from PSIPRED are superior to those from PHD."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f653a770",
   "metadata": {},
   "source": [
    "## Discussion\n",
    "### Similarity and differences between secondary structure prediction methodologies\n",
    "The main similarity among secondary structure prediction methods is that they all aim to predict local structures of proteins, such as α-helices, β-sheets, and coils. These methods typically use the amino acid sequence of the protein as input and rely on various algorithms and statistical models to predict the secondary structure type for each amino acid. Additionally, prediction methods usually employ similar evaluation criteria, such as the Q3 score, to measure prediction accuracy.\n",
    "\n",
    "Different prediction methods utilize different algorithms. Some may use neural networks, some may employ energy minimization modeling, and some might identify homologous proteins through sequence alignment. Due to the use of different algorithms and feature sets, there are variations in the prediction accuracy among these methods.\n",
    "\n",
    "### PSIPRED\n",
    "PSIPRED uses artificial neural network machine learning techniques, analyzing the output from PSI-BLAST through two feed-forward neural networks and employing a cross-validation procedure to evaluate performance. Initially, PSI-BLAST is used to perform multiple rounds of searches on a given protein sequence to generate a Position-Specific Scoring Matrix (PSSM), which describes the probability of each amino acid occurring at every position in the sequence. PSIPRED then processes this PSSM using two cascaded feed-forward neural networks. The first network generates preliminary secondary structure predictions, and the second network further processes these data, taking into account the local sequence environment to optimize the initial predictions. Its output is a confidence score for the secondary structure of each amino acid residue, indicating the probability that the residue belongs to various possible secondary structures.\n",
    "\n",
    "Through Part III calculations, we can determine that PSIPRED has a Q3 score of 87% in this case, indicating its excellent predictive capability.\n",
    "\n",
    "McGuffin, L. J., Bryson, K., & Jones, D. T. (2000). The PSIPRED protein structure prediction server. Bioinformatics (Oxford, England), 16(4), 404–405. https://doi.org/10.1093/bioinformatics/16.4.404\n",
    "\n",
    "### PHD\n",
    "PHD predicts protein secondary structures by combining sequence profile alignment and a neural network system. It begins by aligning sequences consecutively to the guide sequence, then compiles an alignment profile used to align the next sequence. After selecting all sequences with significant homology, the profile is recompiled, and the sequences are aligned consecutively again using dynamic programming. The sequence profile and conservation weight are inputs to a two-layer feed-forward network. The output of this network indicates the probability that the residue is in a helix, strand, or coil. The output from the first layer network, the sequence-to-structure net, serves as the input for the second layer, the structure-to-structure network, which has the same architecture as the first layer. The second layer helps correct deficiencies due to the first layer's focus solely on the central residue and neglect of direct correlations between adjacent residues. Finally, the third layer's output is derived by calculating an arithmetic average over the outputs of several independently trained two-level networks. The core of the method is to provide probabilities for each residue belonging to specific secondary structure categories, then make the final structure prediction based on these probabilities.\n",
    "\n",
    "Through Part III calculations, we can determine that PHD has a Q3 score of 68.5% in this case. Although this is also good, it does not perform as well as PSIPRED.\n",
    "\n",
    "Rost, B., Sander, C., & Schneider, R. (1994). PHD-an automatic mail server for protein secondary structure prediction. Bioinformatics, 10(1), 53–60. https://doi.org/10.1093/bioinformatics/10.1.53\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "399a9fdc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8f75b66",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd41c8dc",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
